L1Trigger/Phase2L1ParticleFlow/src/utils/DiscretePF2Firmware.h

```diff
@@ -0,0 +1,69 @@
+#ifndef FASTPUPPI_NTUPLERPRODUCER_DISCRETEPF2FIRMWARE_H
+#define FASTPUPPI_NTUPLERPRODUCER_DISCRETEPF2FIRMWARE_H
+
+/// NOTE: this include is not standalone, since the path to DiscretePFInputs is different in CMSSW & Vivado_HLS
+
+#include "../firmware/data.h"
+#include <vector>
+
+namespace dpf2fw {
+
+  // convert inputs from discrete to firmware
+  void convert(const l1tpf_impl::PropagatedTrack &in, TkObj &out) {
+    out.hwPt = in.hwPt;
+    out.hwPtErr = in.hwCaloPtErr;
+    out.hwEta = in.hwEta;  // @calo
+    out.hwPhi = in.hwPhi;  // @calo
+    out.hwZ0 = in.hwZ0;
+    out.hwTightQuality = (in.hwStubs >= 6 && in.hwChi2 < 500);
```

Comment:
magic numbers

Reply:
unfortunately I need to use this source file also from environments where I don't have a configuration file.
if a #define is preferrable I can add it somewhere and use it here.

Reply:
#defines should not be used for constants, see 7.6: http://cms-sw.github.io/cms_coding_rules.html